ui/src/_components/Game.svelte

```diff
@@ -1,13 +1,22 @@
 <script>
-    import PlayersScoreboard from "./PlayersScoreboard.svelte"
-    import GameChat from "./GameChat.svelte"
-    import SongTimer from "./SongTimer.svelte"
+	import PlayersScoreboard from './PlayersScoreboard.svelte';
+	import GameChat from './GameChat.svelte';
+	import SongTimer from './SongTimer.svelte';
+	import { onMount } from 'svelte';
+	let menuOpen = false;
 </script>
 
-<div class="grid md:grid-cols-game md:grid-rows-1 grid-cols-2 grid-rows-2 min-h-[80%] m-10">
-    <PlayersScoreboard/>
-    <SongTimer/>
-    <div class="col-span-2 md:col-span-1">
-        <GameChat/>
-    </div>
+<div class="flex flex-col md:flex-row grow w-full">
+	<div class="md:w-3/12 p-4 md:order-first md:sticky md:top-0">
+		<button class="md:hidden" on:click={() => (menuOpen = !menuOpen)}>Toggle Menu</button>
+		<div class={`${menuOpen ? 'block' : 'hidden'} md:block`}>
```

Comment:
this is compacted enought to justify a sprate function

Reply:
????

This is not really about functions ig, this is more of logic set "how a component should behave"